Adds variation to traitor motives

Sniblet opened the pull request:
# Document the changes in your pull request

https://forums.yogstation.net/threads/expand-on-the-last-line-of-the-traitor-alert.24591/
> -Add a greater variety of incentives for traitors.
> -Make the strength of the traitor’s incentive scale to their objectives - it’s fine and good for variety if the incentive is more than strong enough, but lowballing it like we do is just insane.

I don't know how to code but this is an attempt at a foundation of that

# Changelog

:cl:  
rscadd: Traitors will now require a bit more than a small pay bonus to kill the entire station
/:cl:

Sniblet committed:
Update datum_traitor.dm

Sniblet committed:
Update datum_traitor.dm

is this what you want

Sniblet committed:
Update datum_traitor.dm

how about this

Sniblet committed:
Apply suggestions from code review

Co-authored-by: Byemoh <baiomurang@gmail.com>

Sniblet committed:
Update datum_traitor.dm

Sniblet committed:
Apply suggestions from code review

Co-authored-by: Byemoh <baiomurang@gmail.com>

ToasterBiome reviewed (changes requested):
change this stuff plz

Sniblet committed:
Update datum_traitor.dm

moral arithmetic is hard... maybe copying viro's severity system would be better

Sniblet committed:
Update datum_traitor.dm

steal objectives are harder to get away with, they need stronger motivation

ToasterBiome reviewed (changes requested):
more lore is good but it should still tell you your pay modifier.
